lib/urlwatch/handler.py

```diff
@@ -47,16 +47,20 @@ def __init__(self, cache_storage, job):
         self.verb = None
         self.old_data = None
         self.new_data = None
+        self.history_data = {}
         self.timestamp = None
         self.exception = None
         self.traceback = None
         self.tries = 0
         self.etag = None
 
     def load(self):
-        self.old_data, self.timestamp, self.tries, self.etag = self.cache_storage.load(self.job, self.job.get_guid())
+        guid = self.job.get_guid()
+        self.old_data, self.timestamp, self.tries, self.etag = self.cache_storage.load(self.job, guid)
         if self.tries is None:
             self.tries = 0
+        if self.job.compared_versions and self.job.compared_versions > 1:
```

Comment:
Does this need `getattr(self.job, 'compared_versions')` or does it work in all cases (don't remember if we set default values for optional job keys)?

Reply:
This should work. All optional keys are set in `__init__`:
https://github.com/thp/urlwatch/blob/9b289b8f20f492313145cf70246a1166b6bcc52c/lib/urlwatch/jobs.py#L71-L74